src/main/java/io/github/moulberry/notenoughupdates/listener/ItemTooltipRngListener.java

```diff
@@ -0,0 +1,306 @@
+		repoReloadNeeded = true;
+	}
+
+	public void checkUpdateData() {
+		if (repoReloadNeeded) {
+			updateRepoData();
+		}
+	}
+
+	private void updateRepoData() {
+		slayerData.clear();
+		dungeonData.clear();
+
+		JsonObject leveling = Constants.LEVELING;
+		if (!leveling.has("slayer_boss_xp") ||
+			!leveling.has("slayer_highest_tier") ||
+			!leveling.has("slayer_tier_colors") ||
+			!leveling.has("rng_meter_dungeon_score")) {
+			return;
+		}
+
+		List<Integer> slayerExp = new ArrayList<>();
+		for (JsonElement element : leveling.get("slayer_boss_xp").getAsJsonArray()) {
+			slayerExp.add(element.getAsInt());
+		}
+
+		List<String> slayerColors = new ArrayList<>();
+		for (JsonElement element : leveling.get("slayer_tier_colors").getAsJsonArray()) {
+			slayerColors.add(element.getAsString());
+		}
+
+		for (Map.Entry<String, JsonElement> entry : leveling.get("slayer_highest_tier").getAsJsonObject().entrySet()) {
+			String slayerName = entry.getKey();
+			int maxTier = entry.getValue().getAsInt();
+			LinkedHashMap<String, Integer> singleSlayerData = new LinkedHashMap<>();
+			for (int i = 0; i < maxTier; i++) {
+				String name = slayerColors.get(i) + "Tier " + (i + 1);
+				singleSlayerData.put(name, slayerExp.get(i));
+			}
+			slayerData.put(slayerName, singleSlayerData);
+		}
+
+		for (Map.Entry<String, JsonElement> entry : leveling.get("rng_meter_dungeon_score").getAsJsonObject().entrySet()) {
+			String dungeonScore = entry.getKey();
+			int score = entry.getValue().getAsInt();
+			dungeonData.put(dungeonScore, score);
+		}
+
+		repoReloadNeeded = false;
+	}
+
+	private void runsRequired(
+		List<String> toolTip,
+		int having,
+		int needed,
+		boolean nextLineProgress,
+		ItemStack stack
+	) {
+		checkUpdateData();
+		if (repoReloadNeeded) return;
+
+		String openChestName = Utils.getOpenChestName();
+		Map<String, Integer> runsData;
+		String labelPlural;
+		String labelSingular;
+		if (openChestName.contains("Catacombs")) {
+			runsData = dungeonData;
+			labelPlural = "Runs";
+			labelSingular = "Run";
+		} else { // Slayer
+			String[] split = openChestName.split(" ");
+			String slayerName = split[0] + " " + split[1];
+			runsData = slayerData.get(slayerName);
+			labelPlural = "Bosses";
+			labelSingular = "Boss";
+		}
+
+		handleArrowKeys(runsData);
+
+		if (currentSelected >= runsData.keySet().size()) {
+			currentSelected = 0;
+		}
+
+		String name = (String) runsData.keySet().toArray()[currentSelected];
+		int gainPerRun = runsData.get(name);
+
+		int runsNeeded = needed / gainPerRun;
+		int runsHaving = having / gainPerRun;
+		String runsNeededFormat = GuiProfileViewer.numberFormat.format(runsNeeded);
+		String runsHavingFormat = GuiProfileViewer.numberFormat.format(runsHaving);
+
+		String progressString = null;
+		if (nextLineProgress) {
+			progressString = toolTip.remove(toolTip.size() - 1);
+		}
+
+		toolTip.add("§9Stats for " + name + "\u00a79: [\u00a7l\u00a7m< \u00a79Switch\u00a7l\u27a1\u00a79]");
```

Comment:
Inconsistent use of § and \u00a7

Reply:
I've copied this line from [ItemTooltipRender](https://github.com/NotEnoughUpdates/NotEnoughUpdates/blob/master/src/main/java/io/github/moulberry/notenoughupdates/listener/ItemTooltipListener.java#L264)

Should we change both lines to using §?

Reply:
The only reason for \u00a7 is for non UTF-8 compilations (which hopefully we don't have any, and if we do then well have more problems to fix). So yeah id say we fix it so we have easier to read sourcecode

Reply:
Implemented.